otherlibs/stdune/src/fpath.ml

```diff
@@ -187,7 +187,7 @@ let rm_rf fn =
 ;;
 
 let traverse =
-  let rec loop on_file on_dir root stack acc =
+  let rec loop on_file on_dir on_broken_symlink root stack acc =
```

Comment:
```suggestion
let traverse ~on_broken_symlink =
  let rec loop on_file on_dir root stack acc =
```

Given `on_broken_symlink` isn't really part of the loop (it is never changed and the recursive calls always reuses the same one), I'd put it outside of the recursive function arguments. That way one does not have to think whether these get mutated or not (same probably goes for `on_file` and `on_dir`) and one less argument to get wrong.

Reply:
Agreed. I was following suite for the other arguments to this function. I'll rearrange it so only the changing arguments are passed to `loop` (in a separate commit so it's clearly not part of handling broken symlinks).